client/foundries.go

```diff
@@ -1648,18 +1654,18 @@ func (a *Api) FactoryCreateWave(factory string, wave *WaveCreate) error {
 	return err
 }
 
-func (a *Api) FactoryListWaves(factory string, limit uint64) ([]Wave, error) {
-	url := a.serverUrl + "/ota/factories/" + factory + "/waves/?limit=" + strconv.FormatUint(limit, 10)
+func (a *Api) FactoryListWaves(factory string, limit uint64, page uint64) (*WaveList, error) {
+	url := a.serverUrl + "/ota/factories/" + factory + "/waves/?limit=" + strconv.FormatUint(limit, 10) + "&page=" + strconv.FormatUint(page, 10)
 	logrus.Debugf("Listing factory waves %s", url)
 
 	body, err := a.Get(url)
 	if err != nil {
 		return nil, err
 	}
 
-	var resp []Wave
-	err = json.Unmarshal(*body, &resp)
-	return resp, err
+	waves := WaveList{}
```

Comment:
Nitpick: `var waves WaveList` is how we currently do it. I'm not sure about the tiny nuances of either way.

Visually, the biggest difference is that in case of an error the `var` way will return `nil`, but the `:=` way will return a partially initialized object.

Reply:
I do the exact same as for DeviceList to be consistent. But if you want we can change it for this one.

Reply:
That was a nipick, so you can ignore it.